rackcity/views/change_plan_views.py

```diff
@@ -0,0 +1,294 @@
+        )
+    id = data['id']
+    try:
+        existing_change_plan = ChangePlan.objects.get(id=id)
+
+    except ObjectDoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    "Model" + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing asset with id="+str(id)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    for field in data.keys():
+        value = data[field]
+        setattr(existing_change_plan, field, value)
+    try:
+        existing_change_plan.save()
+        return JsonResponse(
+            {
+                "success_message":
+                    Status.SUCCESS.value +
+                    "Change Plan " +
+                    str(existing_change_plan.name) +
+                    " modified",
+                "related_id": str(existing_change_plan.id)
+            },
+            status=HTTPStatus.OK,
+        )
+    except Exception as error:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    parse_save_validation_error(error, "Asset"),
+                "errors": str(error)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def change_plan_add(request):
+    """
+    Add a new change plan
+    """
+    data = JSONParser().parse(request)
+    if 'id' in data:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.CREATE_ERROR.value + GenericFailure.INTERNAL.value,
+                "errors": "Don't include 'id' when creating a change planner"
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    data["owner"] = request.user.id
+    serializer = AddChangePlanSerializer(data=data)
+    if not serializer.is_valid(raise_exception=False):
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.INVALID_INPUT.value +
+                    parse_serializer_errors(serializer.errors),
+                "errors": str(serializer.errors)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    try:
+        change_plan = serializer.save()
+    except Exception as error:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.CREATE_ERROR.value +
+                    parse_save_validation_error(error, "Change Plan"),
+                "errors": str(error)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    return JsonResponse(
+        {
+            "success_message":
+                Status.SUCCESS.value +
+                "Change Plan " +
+                str(change_plan.name) +
+                " created",
+            "related_id": str(change_plan.id)
```

Comment:
what is this being used for?

Reply:
instead of returning the id of the new change plan as part of the success message, it makes it easier on the front end to always just have a separate related_id field if we need to provide a link to the new change plan. not actively using this/would want to do a refactor across other endpoints

Reply:
swaggy